Setup.hs

```diff
@@ -1,11 +1,15 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE TupleSections #-}
 
+#ifndef MIN_VERSION_Cabal
```

Comment:
just in case this doesn't seem obvious: the trick here is `MIN_VERSION_Cabal` is defined and available to custom Setup.hs scripts if either GHC >= 8.0 or cabal-install >= 1.24 is used. So if it isn't defined, it's very likely we don't have Cabal >= 2.0.